core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingRepository.scala

```diff
@@ -93,15 +97,25 @@ class BillingRepository(dataSource: SlickDataSource) {
 
   def failUnlessHasNoWorkspaces(projectName: RawlsBillingProjectName)(implicit ec: ExecutionContext): Future[Unit] =
     dataSource.inTransaction { dataAccess =>
-      dataAccess.workspaceQuery.countByNamespace(projectName) map { count =>
-        if (count == 0) ()
-        else
+      for {
+        workspaces <- dataAccess.workspaceQuery.listWorkspacesByNamespace(projectName)
```

Comment:
Since there's only one query here, the for-comp isn't strictly necessary. You could `.map` `listWorkspacesByNamespace` and then do all the same things within that block. It's a style preference thing though so up to you!

Reply:
Ah, thanks! I got confused because I thought that map would iterate over all the workspaces, which wasn't what I wanted. But I guess it does not because what is actually returned is a `ReadAction`.

I made your suggested change.